### DISCUSSION TITLE
Bash script not running on startup

### DISCUSSION THREAD

**Discussion by kushagharahi** [gtxaspec/wz_mini_hacks#571] [General]
I wrote this small script to feed motion detection alerts to my camera plugin. It works when I manually run it (`sh motionDetection.sh`)

```
#!/opt/wz_mini/bin/bash

ip=$(ip -o -4 addr list wlan0 | awk '{print $4}' | cut -d/ -f1)

case $ip in
  "192.168.1.4")
    cameraName="Kitchen"
    ;;

  "192.168.1.60")
    cameraName="Backyard"
    ;;

  "192.168.1.81")
    cameraName="Garage"
    ;;

  "192.168.1.80")
    cameraName="Living Room"
    ;;

  *)
    echo "Camera name for $ip not found! Exiting."
    exit 1
    ;;
esac

# URL encode cameraName spaces with %20.
cameraName=${cameraName//' '/'%20'}

while true
do
  ret=$(cmd waitmotion 1000 | tr -d '\0')
  #printf "$ret\n"
  if [[ "$ret" == "detect"* ]] ; then
    echo "Reporting motion!"
    curl -X POST "http://192.168.1.101:8088/motion?$cameraName"
    sleep 1
  else
    curl -X POST "http://192.168.1.101:8088/motion/reset?$cameraName"
    echo "No motion found!"
    sleep 1
  fi
done
```

I named it `motionDetection.sh` and did a `chmod +x` to make it executable.

I tried the follow methods of getting it to run on startup:
- Placing it in `opt/wz_mini/etc/rc.local.d/motionDetection.sh`
- Placing it in `media/mmc/wz_mini/etc/rc.local.d/motionDetection.sh`
- Placing it in `opt/wz_mini/etc/configs/motionDetection.sh` and updating the `CUSTOM_SCRIPT_PATH` config with that path (`opt/wz_mini/etc/configs/motionDetection.sh`)

What am I doing wrong? How can I view logs related to this?

**Comment by akivacp** [comment]
Try calling the command in the cron file in `/media/mmc/wz_mini/etc/cron/root`

`@reboot /opt/wz_mini/etc/configs/motionDetection.sh` (or wherever the file is located)

See if that works
